src/query/models/tag.go

```diff
@@ -145,17 +146,47 @@ func (m Matchers) ToTags() (Tags, error) {
 	return Normalize(tags), nil
 }
 
-// ID returns a string representation of the tags
-func (t Tags) ID() string {
-	b := make([]byte, 0, len(t))
+// StringID returns a string representation of the tags
+func (t Tags) StringID() string {
+	var (
+		idLen      = t.IDLen()
+		strBuilder = strings.Builder{}
+	)
+
+	strBuilder.Grow(idLen)
+	for _, tag := range t {
+		strBuilder.WriteString(tag.Name)
+		strBuilder.WriteByte(eq)
+		strBuilder.WriteString(tag.Value)
+		strBuilder.WriteByte(sep)
+	}
+
+	return strBuilder.String()
+}
+
+// WriteBytesID writes out the ID representation
+// of the tags into the provided buffer.
+func (t Tags) WriteBytesID(b []byte) []byte {
```

Comment:
nit: why does this one need a buffer, rather than creating one like the stringID function?

Reply:
The idea is that the caller controls the buffer which enables us to add pooling later without making an API change

Reply:
yeah I like the caller providing a buffer